### PR TITLE
fix(tui): support resizing

### PR DESCRIPTION
**Pull request by rgrinberg**
This PR allows the tui to be rerendered on resize events. This change is twofold:

* Allow handling sigwinch in the tui thread by blocking it in every other thread.
* Calling render whenever we get the winch.

Adding @snowleopard as a reviewer for the insignificant changes to scheduler to allow us to handle sigwinch.

### CODE REVIEW

**Review by Alizter — DISMISSED** [review]

**Review by Alizter — APPROVED** [review]
Resizing now renders reliably.

**Comment by rgrinberg** [comment]
ping @snowleopard

File: src/dune_engine/scheduler.ml [hunk 1]
```diff
@@ -57,7 +57,9 @@ module Shutdown = struct
       | _ -> None)
 end
 
-let blocked_signals : Signal.t list = [ Int; Quit; Term ]
+let interrupt_signals : Signal.t list = [ Int; Quit; Term ]
+
+let blocked_signals : Signal.t list = Winch :: interrupt_signals
```
Comment (snowleopard) [inline src/dune_engine/scheduler.ml:62]: Could you add some comments on the difference between `interrupt_signals` and `blocked_signals`?

File: src/dune_threaded_console/dune_threaded_console.ml [hunk 2]
```diff
@@ -135,7 +136,7 @@ let progress () =
       (* The current console doesn't react to user events so we just sleep until
           the next loop iteration. Because it doesn't react to user input, it cannot
           modify the UI state, and as a consequence doesn't need the mutex. *)
-      let handle_user_events ~now ~time_budget _ =
+      let handle_user_events ~now ~time_budget _mutex _state =
```
Comment (snowleopard) [inline src/dune_threaded_console/dune_threaded_console.ml:139]: Could you add type signatures to ignored values? Just to make sure we notice if the type changes and name becomes stale.

**Review by snowleopard — APPROVED** [review]
Looks good but I left a couple of comments.

**Pull request merged**